plugins/speculation-rules/readme.txt

```diff
@@ -63,6 +63,27 @@ add_filter(
 );
 `
 
+Keep in mind that sometimes it may be useful to exclude a URL from prerendering while still allowing it to be prefetched. For example, a page with client-side JavaScript to update user state should probably not be prerendered, but it would be reasonable to prefetch.
+
+For this purpose, the `plsr_speculation_rules_href_exclude_paths` filter receives the current mode (either "prefetch" or "prerender") to provide conditional exclusions.
+
+The following example would ensure that URLs like `https://example.com/products/...` cannot be prerendered, while still allowing them to be prefetched.
+`
+<?php
+
+add_filter(
+	'plsr_speculation_rules_href_exclude_paths',
+	function ( $exclude_paths, $mode ) {
```

Comment:
Maybe helpful to include typing here at least for a documentation aide:

```suggestion
	static function ( array $exclude_paths, string $mode ): array {
```

Reply:
+1 for the type hints, -1 for the `static` as it's not really a necessity and one more piece of overhead that doesn't need to be there :)